v2/robotmk/src/environment.rs

```diff
@@ -65,16 +64,20 @@ fn build_environment<'a>(
             let drop_suite = matches!(environment_build_status, EnvironmentBuildStatus::Failure(_));
             environment_build_states_administrator
                 .insert_and_write_atomic(&suite.name, environment_build_status)?;
-            drop_suite
+            if drop_suite {
```

Comment:
I think we have to invert this, right? We want to return `Some(suite)` if we the building suceeded.

Reply:
Yes, good catch. Not sure how to test this :thinking: